README.md

```diff
@@ -14,21 +14,98 @@ It is work in progress, but functions well.
-- [x] [go.mod][tree-sitter-go-mod]
-- [ ] HTML
-- [x] [Javascript/JSX][tree-sitter-javascript]
-- [x] [Julia][tree-sitter-julia]
-- [x] [Lua][tree-sitter-lua]
-- [x] [Rust][tree-sitter-rust]
-- [x] [Zig][tree-sitter-zig]
-
-If you want more languages supported, open an issue.
+
+Any language supported by `tree-sitter` should (in theory) be supported as well.
+
+It is possible to use grammar installed locally or directly from their git repository. Additionally 
+it is possible to specify where to find the queries of the language as well as use sub path for projects
+containing more then one language.
+## Default grammars
+
+A few grammars have default configurations, pre-compiled parsers and custom queries:
+
+The supported languages are:
+ 
+- [C][tree-sitter-c]
+- [C++][tree-sitter-cpp]
+- [D][tree-sitter-d]
+- [Diff][tree-sitter-diff]
+- [Go][tree-sitter-go]
+- [go.mod][tree-sitter-go-mod]
+- [Javascript/JSX][tree-sitter-javascript]
+- [Julia][tree-sitter-julia]
+- [Lua][tree-sitter-lua]
+- [Rust][tree-sitter-rust]
+- [Zig][tree-sitter-zig]
+
+To use any of this grammars simply add this to your conifg 
+
+```lua
+local egconfig = require "plugins.evergreen.config"
+
+egconfig.addGrammar {
+  lang = "LANGUAGE",
+  precompiled = true -- optional: default true, if false use default git repository to install grammar
+}
+```
+
+
+## Locally installed grammars
+
+To add a grammar to lite-xl simply add the following configuration to your `init.lua` script:
+
+```lua
+local egconfig = require "plugins.evergreen.config"
+
+egconfig.addGrammar {
+  path = "YOUR GRAMMAR LOCAL PATH",
+  lang = "LANGUAGE NAME",
+  extensions = "FILE EXTENSIONS", -- optional for extension name based
+  filename = "FILE NAME", -- optional for file name based 
+  queries = "QUERIES SUB PATH", -- optional sub path of grammar queries, default = 'queries'
+}
+```
+
+## From git repository 
+To add a grammar directly from a git repository add this to your `init.lua` scripts:
+
+```lua
+local egconfig = require "plugins.evergreen.config"
+egconfig.addGrammar {
+  git = "GIT REPO",
+  lang = "LANGUAGE NAME",
+  extensions = "FILE EXTENSIONS", -- optional for extension name based
+  filename = "FILE NAME", -- optional for file name based 
+  queries = "QUERIES SUB PATH", -- optional sub path of grammar queries, default = 'queries'
+  subpath = "GRAMMAR SUB PATH", -- optional sub path if grammar is nested, default = nil
+  revision = "SPECIFIC GIT REVISION", -- optional version of the code to clone
+}
+```
+
+## Example 
+
+```lua
+local egconfig = require "plugins.evergreen.config"
+
+egconfig.addGrammar {
+  path = "~/.grammars/tree-sitter-epics/epics-db",
+  lang = "epics_db",
+  extensions = "db,vdb",
+}
+
+egconfig.addGrammar {
+  git = "https://github.com/tree-sitter/tree-sitter-cpp.git",
+  lang = "cpp",
+  extensions = "cpp,hpp,cc",
+}
+
+egconfig.addGrammar {
+  git = "https://github.com/tree-sitter/tree-sitter-c.git",
+  lang = "c",
+  extensions = "c,h",
+}
+
```

Comment:
this entire section should be condensed so that the examples will also specify all available values.

Reply:
Now I am using tables for both `extensions` and `filenames`. Now I am missing to rewrite the example section.

Reply:
I simplify the readme and the example section I hope it is now better.